atomic_reactor/plugins/pre_change_from_in_df.py

```diff
@@ -72,17 +76,23 @@ def run(self):
         # enclose images from dfp
         enclosed_parent_images = []
         for df_img in dfp.parent_images:
+            if base_image_is_scratch(df_img):
+                enclosed_parent_images.append(df_img)
+                continue
             parent = ImageName.parse(df_img)
             if organization:
                 parent.enclose(organization)
             enclosed_parent_images.append(parent)
 
         missing = [df_img for df_img in enclosed_parent_images
                    if df_img not in builder.parent_images]
-        if missing:
+        missing_set = set(missing)
+        if 'scratch' in missing_set:
+            missing_set.remove('scratch')
```

Comment:
Maybe we want 'scratch' to be defined in constants.py?

Reply:
will do